src/libraries/System.Private.CoreLib/src/System/Text/UTF32Encoding.cs

```diff
@@ -1101,10 +1101,7 @@ public override int GetMaxCharCount(int byteCount)
                 charCount /= 2;
             }
 
-            if (charCount > 0x7fffffff)
-                throw new ArgumentOutOfRangeException(nameof(byteCount), SR.ArgumentOutOfRange_GetCharCountOverflow);
-
-            return (int)charCount;
+            return charCount;
```

Comment:
This is actually a bug in the original code. On line 1091 above, _charCount_ should be typed as _long_, not _int_.